packages/inventory/src/components/table/EntityTableToolbar.js

```diff
@@ -0,0 +1,327 @@
+/* eslint-disable camelcase */
+/* eslint-disable no-unused-vars */
+import React, { Fragment, useEffect, useCallback, useReducer } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components/components/esm/Skeleton';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/components/esm/PrimaryToolbar';
+import { fetchAllTags, clearFilters, entitiesLoading, toggleTagModal } from '../../redux/actions';
+import debounce from 'lodash/debounce';
+import flatMap from 'lodash/flatMap';
+import {
+    TagsModal,
+    mapGroups,
+    TEXT_FILTER,
+    reduceFilters,
+    TEXTUAL_CHIP,
+    STALE_CHIP,
+    REGISTERED_CHIP,
+    TAG_CHIP,
+    arrayToSelection,
+    loadSystems
+} from '../../shared';
+import { onDeleteFilter, onDeleteTag } from './helpers';
+import {
+    useStalenessFilter,
+    useTextFilter,
+    useRegisteredWithFilter,
+    useTagsFilter,
+    textFilterState,
+    textFilterReducer,
+    filtersReducer,
+    stalenessFilterReducer,
+    stalenessFilterState,
+    registeredWithFilterReducer,
+    registeredWithFilterState,
+    tagsFilterReducer,
+    tagsFilterState
+} from '../filters';
+
+/**
+ * Table toolbar used at top of inventory table.
+ * It uses couple of filters and acces redux data along side all passed props.
+ * @param {*} props used in this component.
+ */
+const EntityTableToolbar = ({
+    total,
+    page,
+    perPage,
+    filterConfig,
+    hasItems,
+    children,
+    actionsConfig,
+    onRefresh,
+    hasCheckbox,
+    activeFiltersConfig,
+    showTags,
+    isLoaded,
+    items,
+    sortBy,
+    ...props
+}) => {
+    const dispatch = useDispatch();
+    const reducer = useReducer(filtersReducer([
+        textFilterReducer,
+        stalenessFilterReducer,
+        registeredWithFilterReducer,
+        tagsFilterReducer
+    ]), {
+        ...textFilterState,
+        ...stalenessFilterState,
+        ...registeredWithFilterState,
+        ...tagsFilterState
+    });
+    const filters = useSelector(({ entities: { activeFilters } }) => activeFilters || []);
+    const loaded = useSelector(({ entities: { loaded } }) => hasItems && isLoaded !== undefined ? (isLoaded && loaded) : loaded);
+    const allTagsLoaded = useSelector(({ entities: { allTagsLoaded } }) => allTagsLoaded);
```

Comment:
^^ all of these selectors with primitive selector type could use the shallow equal hook.